### DISCUSSION TITLE
Is it possible to create a new user using migrations?

### DISCUSSION THREAD

**Discussion by aashishvasu** [pocketbase/pocketbase#2219] [Q&A] [answered]
I am currently using the method suggested in #1354 to create a new admin user. Is it possible to create a normal user in a migration as well?

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to create an app user with a JS migration you can use something like:

```js
migrate((db) => {
  const dao = new Dao(db)

  const collection = dao.findCollectionByNameOrId("users")

  const record = new Record(collection)
  record.setUsername("john.doe")
  record.setEmail("john.doe@example.com")
  record.setPassword("123456")
  record.setVerified(true)
  record.set("someOtherRegularField", 123)

  return dao.saveRecord(record)
}, (db) => {
  const dao = new Dao(db)

  const record = dao.findAuthRecordByEmail("users", "john.doe@example.com")

  return dao.deleteRecord(record)
})
```

_The JS api will be documented together with the planned JS app event hooks in the near future (most likely in 0.16/0.17). You can also check https://github.com/pocketbase/pocketbase/discussions/1464._

**Reply by aashishvasu** [reply]
Thank you! This works super good
